To: vim_dev@googlegroups.com
Subject: Patch 7.3.184
Fcc: outbox
From: Bram Moolenaar <Bram@moolenaar.net>
Mime-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit
------------

Patch 7.3.184
Problem:    Static code analysis errors in riscOS.
Solution:   Make buffer size bigger. (Dominique Pelle)
Files:	    src/gui_riscos.c


*** ../vim-7.3.183/src/gui_riscos.c	2010-08-15 21:57:28.000000000 +0200
--- src/gui_riscos.c	2011-05-10 15:49:56.000000000 +0200
***************
*** 1756,1762 ****
  
  	if (button & 0x444)
  	{
! 	    int	    front_block[10];
  	    /* Dragging with Select - bring window to front first */
  	    front_block[0] = gui.window_handle;
  	    swi(Wimp_GetWindowState, 0, front_block);
--- 1756,1762 ----
  
  	if (button & 0x444)
  	{
! 	    int	    front_block[64];
  	    /* Dragging with Select - bring window to front first */
  	    front_block[0] = gui.window_handle;
  	    swi(Wimp_GetWindowState, 0, front_block);
***************
*** 1874,1880 ****
  
      if (ro_dragging == DRAG_RESIZE_WINDOW)
      {
! 	/* Resizeing the main window. */
  	block[2] = y;
  	block[3] = x;
  	ro_open_main(block);
--- 1874,1880 ----
  
      if (ro_dragging == DRAG_RESIZE_WINDOW)
      {
! 	/* Resizing the main window. */
  	block[2] = y;
  	block[3] = x;
  	ro_open_main(block);
***************
*** 2651,2657 ****
      long_u	length;
  
      block[0] = 48;			/* Size of block. */
!     block[3] = 0;			/* Orinial message. */
      block[4] = 0x10;			/* Data request. */
      block[5] = gui.window_handle;
      block[6] = RO_LOAD_CLIPBOARD;	/* Internal handle. */
--- 2651,2657 ----
      long_u	length;
  
      block[0] = 48;			/* Size of block. */
!     block[3] = 0;			/* Original message. */
      block[4] = 0x10;			/* Data request. */
      block[5] = gui.window_handle;
      block[6] = RO_LOAD_CLIPBOARD;	/* Internal handle. */
*** ../vim-7.3.183/src/version.c	2011-05-10 15:41:59.000000000 +0200
--- src/version.c	2011-05-10 15:51:29.000000000 +0200
***************
*** 716,717 ****
--- 716,719 ----
  {   /* Add new patch number below this line */
+ /**/
+     184,
  /**/

-- 
If bankers can count, how come they have eight windows and
only four tellers?

 /// Bram Moolenaar -- Bram@Moolenaar.net -- http://www.Moolenaar.net   \\\
///        sponsor Vim, vote for features -- http://www.Vim.org/sponsor/ \\\
\\\  an exciting new programming language -- http://www.Zimbu.org        ///
 \\\            help me help AIDS victims -- http://ICCF-Holland.org    ///
